File: VM/HotFixDemo/app/src/main/java/com/tkm/hotfixdemo/activities/MainActivity.java
package com.tkm.hotfixdemo.activities;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.widget.TextView;

import com.tkm.hotfixdemo.R;
import com.tkm.hotfixdemo.utils.Calculator;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tvShow = findViewById(R.id.tv_show);
        TextView tvVersion = findViewById(R.id.tv_version);

        tvShow.setText(Calculator.showResult());
        tvVersion.setText(Calculator.version());
    }
}

File: JetPack/LifeCycle/app/src/main/java/com/tkm/lifecycle/usecase1/MainActivity.kt
package com.tkm.lifecycle.usecase1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.tkm.lifecycle.R

class MainActivity : AppCompatActivity() {

    private val listener = MyListener()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        listener.create()
    }

    override fun onStart() {
        super.onStart()
        listener.start()
    }

    override fun onResume() {
        super.onResume()
        listener.resume()
    }

    override fun onPause() {
        super.onPause()
        listener.pause()
    }

    override fun onStop() {
        super.onStop()
        listener.stop()
    }

    override fun onDestroy() {
        super.onDestroy()
        listener.destroy()
    }
}